aws/hll/aws-s3-transfer-manager/src/download/header.rs

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use core::fmt;
+use std::str::FromStr;
+
+use crate::error;
+
+/// Representation of `Range` header.
+/// NOTE: S3 only supports a single bytes range this is a simplified representation
+#[derive(Debug, Clone, PartialEq)]
+pub(crate) struct Range(pub(crate) ByteRange);
```

Comment:
Since `Range` is already a type in `std` that seems like it might be used at some point in this crate should we call this `RangeHeader`?

Reply:
it's _likely_ only used during discovery and we can disambiguate by either only referring to it as `header::Range` (which I believe we do know) or giving the import an alias (e.g. `use ... as <alias>`)

Reply:
(that being said if you feel strongly I can change it makes no big difference to me)

Reply:
As long as it's `pub(crate)`, it's a two-way door